src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -201,15 +201,21 @@ public sealed class Range
             public Range() { }
 
             public Range(int index)
-                : this(index, index) { }
+            {
+                Contracts.CheckParam(index >= 0, nameof(index), "Must be non-negative");
+                Min = index;
+                Max = index;
+            }
 
-            public Range(int min, int max)
+            public Range(int min, int? max)
             {
-                Contracts.CheckParam(min >= 0, nameof(min), "min must be non-negative.");
-                Contracts.CheckParam(max >= min, nameof(max), "max must be greater than or equal to min.");
+                Contracts.CheckParam(min >= 0, nameof(min), "Must be non-negative");
+                Contracts.CheckParam(!(max < min), nameof(max), "If specified, must be greater than or equal to " + nameof(min));
 
                 Min = min;
                 Max = max;
+                ForceVector = true;
```

Comment:
This feels wrong for `ForceVector` to get a different default value than what is given when the default constructor is called.

Reply:
My revision is correct and the old code was wrong. If someone wants something to be a scalar, they call the constructor with *one* index, and in the case of wanting a vector, they call the construct with a *range* of indices. If those indices in the latter case *happened* to just span 1 element, then the resulting column would be a scalar, which I inadvertently suffered from and which is rather surprising.